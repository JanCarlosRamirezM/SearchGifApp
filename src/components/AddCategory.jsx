import { useState } from "react";

export const AddCategory = ({ addNewCategory }) => {
  const [search, setSearch] = useState("");

  const handleChange = (event) => {
    setSearch(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    const category = search.trim();

    if (category.length === 0) return;

    addNewCategory(category);
    setSearch("");
  };

  return (
    <form
      onSubmit={handleSubmit}
      className="row justify-content-center containerInputSearch "
    >
      <input
        type="text"
        id="inputSearch"
        placeholder="Busca tus Gifs Favoritos"
        value={search}
        onChange={handleChange}
      />
    </form>
  );
};
