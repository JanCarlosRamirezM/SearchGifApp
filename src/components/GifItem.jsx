export const GifItem = ({ url, name, title }) => {
  return (
    <div className="col">
      <div className="card">
        <img src={url} className="card-img-top" alt={name} />
        <div className="card-body">
          <h5 className="card-title">{title}</h5>
        </div>
      </div>
    </div>
  );
};
