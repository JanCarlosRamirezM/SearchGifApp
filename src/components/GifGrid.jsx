import { GifItem } from "./GifItem";


import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

  const [images] = useFetchGifs(category);

  return (
    <>
      <div className="containerGifGrid" key={category}>
        <h2 className="mb-3">{category}</h2>

        <div className="row row-cols-1 row-cols-md-2 row-cols-lg-5 g-4">
          {images.map((image) => (
            <GifItem key={image.id} {...image} />
          ))}
        </div>
      </div>
    </>
  );
};
