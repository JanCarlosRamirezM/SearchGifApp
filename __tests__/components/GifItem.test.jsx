import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Test in <GifItem/>", () => {
  const props = {
    title: "Batman",
    name: "Batman",
    url: "https://tenor.com/search/batman-gifs",
  };
  it("debe hacer match con el snapshot ", () => {
    const { container } = render(<GifItem {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("debe mostrar la url y el alt de la imagen", () => {
    render(<GifItem {...props} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(props.url);
    expect(alt).toBe(props.name);
  });

  it("buscar el titulo del itemn", () => {
    render(<GifItem {...props} />);
    expect(screen.getByText("Batman")).toBeTruthy();
  });
});
