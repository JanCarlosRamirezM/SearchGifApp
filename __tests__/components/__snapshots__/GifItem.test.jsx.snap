// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="col"
  >
    <div
      class="card"
    >
      <img
        alt="Batman"
        class="card-img-top"
        src="https://tenor.com/search/batman-gifs"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Batman
        </h5>
      </div>
    </div>
  </div>
</div>
`;
